nat run reaches an unexpected state: "AssertionError: Should not reach here. Should have been caught by pre_run"

### Version

0.4.3

### Which installation method(s) does this occur on?

PyPi

### Describe the bug.

When I do simple:

```shell
nat run --config_file workflow.yaml
```

I get 

```shell
Error: Should not reach here. Should have been caught by pre_run
```

Based on code, I see that pre_run is expected to raise a meaningful error, but its not called for some reason.

### Minimum reproducible example

```shell
nat run --config_file workflow.yaml
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

2025-09-24 14:01:39,977 - nat.cli.commands.start - ERROR - Failed to initialize workflow
Traceback (most recent call last):
  File "/Users/anatolii.stehnii/workspace/buzok_experiments/new_agent_architecture/nat_example/.venv/lib/python3.11/site-packages/nat/cli/commands/start.py", line 225, in invoke_subcommand
    return asyncio.run(run_plugin())
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/anatolii.stehnii/workspace/buzok_experiments/new_agent_architecture/nat_example/.venv/lib/python3.11/site-packages/nest_asyncio.py", line 30, in run
    return loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/anatolii.stehnii/workspace/buzok_experiments/new_agent_architecture/nat_example/.venv/lib/python3.11/site-packages/nest_asyncio.py", line 98, in run_until_complete
    return f.result()
           ^^^^^^^^^^
  File "/Users/anatolii.stehnii/.local/share/uv/python/cpython-3.11.11-macos-aarch64-none/lib/python3.11/asyncio/futures.py", line 203, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "/Users/anatolii.stehnii/.local/share/uv/python/cpython-3.11.11-macos-aarch64-none/lib/python3.11/asyncio/tasks.py", line 277, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "/Users/anatolii.stehnii/workspace/buzok_experiments/new_agent_architecture/nat_example/.venv/lib/python3.11/site-packages/nat/cli/commands/start.py", line 223, in run_plugin
    await front_end_plugin.run()
  File "/Users/anatolii.stehnii/workspace/buzok_experiments/new_agent_architecture/nat_example/.venv/lib/python3.11/site-packages/nat/front_ends/simple_base/simple_front_end_plugin_base.py", line 50, in run
    await self.run_workflow(session_manager)
  File "/Users/anatolii.stehnii/workspace/buzok_experiments/new_agent_architecture/nat_example/.venv/lib/python3.11/site-packages/nat/front_ends/console/console_front_end_plugin.py", line 93, in run_workflow
    assert False, "Should not reach here. Should have been caught by pre_run"
           ^^^^^
AssertionError: Should not reach here. Should have been caught by pre_run

</pre></details>


### Other/Misc.

Also, its `--input`, not `--input_query` according to `--help`.

### Code of Conduct

- [x] I agree to follow the NeMo Agent toolkit Code of Conduct
- [x] I have searched the [open bugs](https://github.com/NVIDIA/NeMo-Agent-Toolkit/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/src/nat/front_ends/console/console_front_end_plugin.py
+++ b/src/nat/front_ends/console/console_front_end_plugin.py
@@ -55,9 +55,10 @@ class ConsoleFrontEndPlugin(SimpleFrontEndPluginBase[ConsoleFrontEndConfig]):
         self.auth_flow_handler = ConsoleAuthenticationFlowHandler()
 
     async def pre_run(self):
-
-        if (not self.front_end_config.input_query and not self.front_end_config.input_file):
-            raise click.UsageError("Must specify either --input_query or --input_file")
+        if (self.front_end_config.input_query is not None and self.front_end_config.input_file is not None):
+            raise click.UsageError("Must specify either --input or --input_file, not both")
+        if (self.front_end_config.input_query is None and self.front_end_config.input_file is None):
+            raise click.UsageError("Must specify either --input or --input_file")
 
     async def run_workflow(self, session_manager: SessionManager):
 

--- a/src/nat/front_ends/simple_base/simple_front_end_plugin_base.py
+++ b/src/nat/front_ends/simple_base/simple_front_end_plugin_base.py
@@ -35,6 +35,8 @@ class SimpleFrontEndPluginBase(FrontEndBase[FrontEndConfigT], ABC):
 
     async def run(self):
 
+        await self.pre_run()
+
         # Must yield the workflow function otherwise it cleans up
         async with WorkflowBuilder.from_config(config=self.full_config) as builder:
 

